src/bgg/norm_simulator.rs

```diff
@@ -11,27 +11,37 @@ impl PolyCircuit {
         &self,
         dim: u32,
         base_bits: u32,
-        unpacked_input_size: usize,
+        packed_input_norms: &[BigUint],
     ) -> NormBounds {
         let one = NormSimulator::new(
             MPolyCoeffs::new(vec![BigUint::one()]),
             BigUint::one(),
             dim,
             base_bits,
         );
-        let mut inputs = vec![];
-        let mut remaining_inputs = unpacked_input_size;
-        let n = dim as usize;
-        while remaining_inputs > 0 {
-            let num_bits = if remaining_inputs >= n { n } else { remaining_inputs };
-            inputs.push(NormSimulator::new(
-                MPolyCoeffs::new(vec![BigUint::one()]),
-                BigUint::one(),
-                dim,
-                base_bits,
-            ));
-            remaining_inputs -= num_bits;
-        }
+        let inputs = packed_input_norms
+            .into_iter()
+            .map(|plaintext_norm| {
+                NormSimulator::new(
+                    MPolyCoeffs::new(vec![BigUint::one()]),
+                    plaintext_norm.clone(),
```

Comment:
https://github.com/MachinaIO/diamond-io/pull/99/files#r2049939353 so we don't need to clone on here